<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PZST::PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>602</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="windowIcon">
   <iconset resource="PZST.qrc">
    <normaloff>:/Icons/configure.png</normaloff>:/Icons/configure.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QWidget" name="widget_3" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="margin">
       <number>1</number>
      </property>
      <item>
       <widget class="QTreeWidget" name="pageSelector">
        <property name="minimumSize">
         <size>
          <width>180</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>180</width>
          <height>16777215</height>
         </size>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string>Page</string>
         </property>
        </column>
        <item>
         <property name="text">
          <string>Editor</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Compiler</string>
         </property>
         <item>
          <property name="text">
           <string>Options</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Search paths</string>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Environment</string>
         </property>
         <item>
          <property name="text">
           <string>Language</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Key bindings</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Color scheme</string>
          </property>
         </item>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="pages">
        <property name="currentIndex">
         <number>5</number>
        </property>
        <widget class="QWidget" name="pageEditor">
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Font</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QFontComboBox" name="fontName">
            <property name="editable">
             <bool>false</bool>
            </property>
            <property name="fontFilters">
             <set>QFontComboBox::MonospacedFonts|QFontComboBox::ScalableFonts</set>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="fontSize">
            <property name="minimum">
             <number>6</number>
            </property>
            <property name="maximum">
             <number>48</number>
            </property>
            <property name="value">
             <number>12</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Font size</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="fontSample">
            <property name="text">
             <string>012345 abcdefg фывапро</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Font sample</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="tabSize">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Tab size</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QCheckBox" name="tabsToSpaces">
            <property name="text">
             <string>Convert tabs to spaces</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="tabsVisible">
            <property name="text">
             <string>Visible tabs</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="UTF16">
            <property name="text">
             <string>Save as UTF-16</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QCheckBox" name="lineNumbers">
            <property name="text">
             <string>Line numbers</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QCheckBox" name="curLine">
            <property name="text">
             <string>Highlight current line</string>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <widget class="QWidget" name="widget_2" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="margin">
              <number>1</number>
             </property>
            </layout>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QCheckBox" name="zebra">
            <property name="text">
             <string>Highlight Spin blocks</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageCompilerOptions">
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="17" column="0" rowspan="2" colspan="3">
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="13" column="1">
           <widget class="QCheckBox" name="genInfo">
            <property name="text">
             <string>Generate Info messages</string>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <widget class="QCheckBox" name="genWarnings">
            <property name="text">
             <string>Generate Warning messages</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QCheckBox" name="nonParallaxExt">
            <property name="text">
             <string>Non-Parallax compatible extensions</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QCheckBox" name="fastConstants">
            <property name="text">
             <string>Fast Constants</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QCheckBox" name="genericSafe">
            <property name="text">
             <string>Generic &quot;Safe&quot; optimizations</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QCheckBox" name="foldUnary">
            <property name="text">
             <string>Fold Constant/Unary pairs</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="eliminateUnused">
            <property name="text">
             <string>Eliminate unused SPIN methods</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="foldConstants">
            <property name="text">
             <string>Fold constants</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Port</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="portName">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Encoding</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="encoding"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageSearchPaths">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QListWidget" name="searchPaths"/>
          </item>
          <item>
           <widget class="QWidget" name="widget" native="true">
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QPushButton" name="pathAdd">
               <property name="text">
                <string>Add</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pathRemove">
               <property name="text">
                <string>Remove</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="moveUp">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Move up</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="moveDown">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Move down</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageMisc">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Language</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="language">
            <item>
             <property name="text">
              <string>English</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Russian</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Changing the language requires program restart</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="margin">
           <number>1</number>
          </property>
          <item>
           <widget class="QTreeWidget" name="shortcuts">
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="rootIsDecorated">
             <bool>false</bool>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <attribute name="headerShowSortIndicator" stdset="0">
             <bool>true</bool>
            </attribute>
            <attribute name="headerShowSortIndicator" stdset="0">
             <bool>true</bool>
            </attribute>
            <column>
             <property name="text">
              <string>Command</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Shortcut</string>
             </property>
            </column>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_4" native="true">
            <layout class="QGridLayout" name="gridLayout_4">
             <property name="margin">
              <number>1</number>
             </property>
             <item row="0" column="0">
              <widget class="PZST::ShortcutEdit" name="first">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="PZST::ShortcutEdit" name="second">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="PZST::ShortcutEdit" name="third">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QToolButton" name="revert">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>Revert</string>
               </property>
               <property name="text">
                <string>...</string>
               </property>
               <property name="icon">
                <iconset resource="PZST.qrc">
                 <normaloff>:/Icons/undo.png</normaloff>:/Icons/undo.png</iconset>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QToolButton" name="clear">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>Clear</string>
               </property>
               <property name="text">
                <string>...</string>
               </property>
               <property name="icon">
                <iconset resource="PZST.qrc">
                 <normaloff>:/Icons/clear.png</normaloff>:/Icons/clear.png</iconset>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QVBoxLayout" name="verticalLayout_4" stretch="40,10,50,0">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QWidget" name="widget_6" native="true">
            <layout class="QGridLayout" name="gridLayout_5">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Foreground</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>Background</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QListWidget" name="foreList">
               <item>
                <property name="text">
                 <string>Condition</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Comment</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Reserved word</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Identifier</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Number</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Type</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>String</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Preprocessor directive</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Operator</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Line numbers</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Selection</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QListWidget" name="backList">
               <item>
                <property name="text">
                 <string notr="true">CON</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">PUB</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">PRI</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">OBJ</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">DAT</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">VAR</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Line numbers</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Current line</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Selection</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_5" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="foreSample">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="backSample">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="PZST::SpinEditor" name="editorSample" native="true"/>
          </item>
          <item>
           <widget class="QCheckBox" name="blockHighlight">
            <property name="text">
             <string>Highlight blocks of code</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PZST::ShortcutEdit</class>
   <extends>QLineEdit</extends>
   <header>shortcutedit.h</header>
  </customwidget>
  <customwidget>
   <class>PZST::SpinEditor</class>
   <extends>QWidget</extends>
   <header>spineditor.h</header>
   <container>1</container>
   <slots>
    <slot>setZebra(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="PZST.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blockHighlight</sender>
   <signal>toggled(bool)</signal>
   <receiver>editorSample</receiver>
   <slot>setZebra(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>256</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
